### PR TITLE
Explain flow in the README + template

### PR DESCRIPTION
**Pull request by maiqueb**
Signed-off-by: Miguel Duarte Barroso <mdbarroso@redhat.com>

### CODE REVIEW

**Comment by maiqueb** [comment]
@jcaamano Please review.

**Comment by jcaamano** [comment]
I think I am missing an `# About` section that gives a short overall description of what the demo is about. And then I would do a more top-down approach that flows from that description on the `# How it works` section. Something like:

# About
This demo showcases the use of a macvtap network backend for VMs using kubevirt....bla bla bla

# How it works
On this demo we spawn two vms using a multus network backend and a novel macvtap frontend ...bla bla

The multus network attachment definition uses macvtap cni and a resource exposed by the macvtap device plugin to provide a macvtap interface that can be used by the VMs...bla bla bla

The kubevirt macvtap frontend is a new bind mechanism that wires the macvtap interface to the VM...bla bla bla

The overall workflow is
- kubevirt takes the resource in the network attachment definition and places it as a resource reques in the pod definition.
- When the pod is spawned, kubelet request the resource to the macvtap device plugin.
- The macvtap device plugin allocates a new macvtap interface and provides access from the pod to the associated /dev/tapX character device.
- Multus obtains the macvtap resource allocated to the pod from kubelet, which is the macvtap interface name, and passes it on to macvtap cni as DeviceID argument.
- The macvtap cni makes this interface available in the pods namespace and performs further configuration on it.
- When running the vm, libvirt is provided with the macvtap interface name, form which it obtains the associated tap device which is passed on to qemu.

What do you think?

File: README.md [hunk 1]
```diff
@@ -1,2 +1,50 @@
...
+
+The overall flow happens as follows:
+  - KubeVirt takes the resource in the network attachment definition and places
+    it as a resource request in the pod definition.
+  - before the pod is spawned, kubelet requests the resource to the macvtap
+    device plugin.
+  - the macvtap device plugin allocates a new macvtap interface, and grants the
+    pod access to the corresponding /dev/tapX character device - via cgroups.
+  - KubeVirt generates the domxml libvirt will require to plug the character
+    device into the VM as a virtio interface.
```
Comment (jcaamano) [inline README.md]: I would probably merge this point into the last, for sake of simplicity.

**Review by jcaamano — APPROVED** [review]
Nice! It's ok for me.

File: README.md (hunk 1 above)
Comment (maiqueb) [reply README.md]: Done.

**Pull request merged**